test/dlc_tests/sanity/test_pre_release.py

```diff
@@ -533,22 +534,34 @@ def test_pip_check(image):
         rf"^tensorflow-serving-api{gpu_suffix} \d\.\d+\.\d+ requires "
         rf"tensorflow{gpu_suffix}, which is not installed.$"
     )
+    allowed_exception_list.append(allowed_tf_exception)
+
     allowed_smclarify_exception = re.compile(
         r"^aiobotocore \d+(\.\d+)* has requirement botocore<\d+(\.\d+)*,>=\d+(\.\d+)*, "
         r"but you have botocore \d+(\.\d+)*\.$"
     )
+    allowed_exception_list.append(allowed_smclarify_exception)
 
     # The v0.22 version of tensorflow-io has a bug fixed in v0.23 https://github.com/tensorflow/io/releases/tag/v0.23.0
     allowed_habana_tf_exception = re.compile(
         rf"^tensorflow-io 0.22.0 requires "
         rf"tensorflow, which is not installed.$"
     )
+    allowed_exception_list.append(allowed_habana_tf_exception)
+
+    # The v0.21 version of tensorflow-io has a bug fixed in v0.23 https://github.com/tensorflow/io/releases/tag/v0.23.0
+    if "tensorflow" in image and "2.6.3" in image:
+        allowed_tf263_exception = re.compile(
+            rf"^tensorflow-io 0.21.0 requires "
+            rf"tensorflow, which is not installed.$"
+        )
+        allowed_exception_list.append(allowed_tf263_exception)
 
     # Add null entrypoint to ensure command exits immediately
     output = ctx.run(
         f"docker run --entrypoint='' {image} pip check", hide=True, warn=True)
     if output.return_code != 0:
-        if not (allowed_tf_exception.match(output.stdout) or allowed_smclarify_exception.match(output.stdout) or allowed_habana_tf_exception.match(output.stdout)):
+        if not(any([allowed_exception.match(output.stdout) for  allowed_exception in allowed_exception_list])):
```

Comment:
Thanks for updating this into a list comprehension form to make it scalable.

Reply:
Thanks @saimidu